cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/OracleGenerator.scala

```diff
@@ -0,0 +1,408 @@
+        queryResult += s"to_date('${createTime}','YYYY-MM-DD HH24:MI:SS')"
+    }
+    s"""
+       |merge into $quote${q.dataset}$quote d
+       |using (select ${queryResult(0)} as "name", 2 as "schema" from dual ) s
+       |on (d."name" = s."name")
+       |when not matched then
+       |insert (${quote}name${quote},${quote}schema${quote},${quote}dataInterval${quote},${quote}stats${quote},${quote}stats.createTime$quote)
+       |values (${queryResult(0)},${queryResult(1)},${queryResult(2)},${queryResult(3)},${queryResult(4)})
+       |when matched then
+       |update
+       |set
+       |  "schema" = ${queryResult(1)},
+       |  "dataInterval" = ${queryResult(2)},
+       |  "stats" = ${queryResult(3)},
+       |  "stats.createTime" = ${queryResult(4)}
+
+       |""".stripMargin
+
+  }
+
+  protected def initExprMap(dataset: String, schemaMap: Map[String, AbstractSchema]): Map[String, FieldExpr] = {
+    val schema = schemaMap(dataset)
+    schema.fieldMap.mapValues {
+      f => FieldExpr(s"$sourceVar.$quote${f.name}$quote", s"$sourceVar.$quote${f.name}$quote")
+    }
+  }
+
+  protected def parseTextRelation(filter: FilterStatement, fieldExpr: String): String = {
+    val wordsArr = ArrayBuffer[String]()
+    filter.values.foreach(w => wordsArr += w.toString)
+    val sb = new StringBuilder(s"${fullTextMatch(0)}($fieldExpr, '")
+    sb.append(wordsArr.mkString(" and ") + s"',1)>0")
+    sb.toString()
+  }
+
+
+  override protected def parseTimeRelation(filter: FilterStatement,
+                                           fieldExpr: String): String = {
+
+
+
+    filter.relation match {
+      case Relation.inRange => {
+        filter.field.dataType match{
+          case time: DataType.Time.type =>
+            s"$fieldExpr >= to_date('${TimeField.TimeFormatForSQL.print(new DateTime(filter.values(0).toString)).split("\\.")(0)}','YYYY-MM-DD HH24:MI:SS') and $fieldExpr < to_date('${TimeField.TimeFormatForSQL.print(new DateTime(filter.values(1).toString)).split("\\.")(0)}','YYYY-MM-DD HH24:MI:SS')"
+          case others =>
+            s"$fieldExpr >= '${filter.values(0)}' and $fieldExpr < '${filter.values(1)}'"
+        }
+      }
+      case _ => {
+        filter.field.dataType match {
+          case time: DataType.Time.type =>
+            s"$fieldExpr ${filter.relation} to_date('${TimeField.TimeFormatForSQL.print(new DateTime(filter.values(0).toString)).split("\\.")(0)}','YYYY-MM-DD HH24:MI:SS')"
+          case _ =>
+            s"$fieldExpr ${filter.relation} '${filter.values(0)}'"
+        }
+      }
+    }
+  }
+
+  //TODO: unnest
+  protected def parseUnnest(unnest: Seq[UnnestStatement],
+                            exprMap: Map[String, FieldExpr], queryBuilder: StringBuilder): ParsedResult = {
+    //return the empty result & exprMap for next step's process.
+    ParsedResult((new ListBuffer[String]), exprMap)
+  }
+  override protected def timeUnitFuncMap(unit: TimeUnit.Value): String = unit match {
```

Comment:
Isn't it the same as the definition in `SQLGenerator`? If so why do we need to override it?